nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/test/java/org/apache/nifi/hbase/TestScanHBase.java

```diff
@@ -0,0 +1,375 @@
+        runner.setProperty(ScanHBase.END_ROW, "row1");
+
+        runner.run();
+        runner.assertTransferCount(ScanHBase.REL_FAILURE, 0);
+        runner.assertTransferCount(ScanHBase.REL_SUCCESS, 0);
+        runner.assertTransferCount(ScanHBase.REL_ORIGINAL, 0);
+
+        Assert.assertEquals(0, hBaseClientService.getNumScans());
+    }
+
+    @Test
+    public void testInvalidTableName() {
+        runner.setProperty(ScanHBase.TABLE_NAME, "${hbase.table}");
+        runner.setProperty(ScanHBase.START_ROW, "row1");
+        runner.setProperty(ScanHBase.END_ROW, "row1");
+
+        runner.enqueue("trigger flow file");
+        runner.run();
+
+        runner.assertTransferCount(ScanHBase.REL_FAILURE, 1);
+        runner.assertTransferCount(ScanHBase.REL_SUCCESS, 0);
+        runner.assertTransferCount(ScanHBase.REL_ORIGINAL, 0);
+
+        Assert.assertEquals(0, hBaseClientService.getNumScans());
+    }
+
+    @Test
+    public void testResultsNotFound() {
+        runner.setProperty(ScanHBase.TABLE_NAME, "table1");
+        runner.setProperty(ScanHBase.START_ROW, "row1");
+        runner.setProperty(ScanHBase.END_ROW, "row1");
+
+        runner.enqueue("trigger flow file");
+        runner.run();
+
+        runner.assertTransferCount(ScanHBase.REL_FAILURE, 0);
+        runner.assertTransferCount(ScanHBase.REL_SUCCESS, 0);
+        runner.assertTransferCount(ScanHBase.REL_ORIGINAL, 1);
+
+        MockFlowFile flowFile = runner.getFlowFilesForRelationship(ScanHBase.REL_ORIGINAL).get(0);
+        flowFile.assertAttributeEquals("scanhbase.results.found", Boolean.FALSE.toString());
+
+        Assert.assertEquals(1, hBaseClientService.getNumScans());
+    }
+
+    @Test
+    public void testScanToContentWithStringValues() {
+        final Map<String, String> cells = new HashMap<>();
+        cells.put("cq1", "val1");
+        cells.put("cq2", "val2");
+
+        final long ts1 = 123456789;
+        hBaseClientService.addResult("row1", cells, ts1);
+        hBaseClientService.addResult("row2", cells, ts1);
+
+        runner.setProperty(ScanHBase.TABLE_NAME, "table1");
+        runner.setProperty(ScanHBase.START_ROW, "row1");
+        runner.setProperty(ScanHBase.END_ROW, "row2");
+        runner.setProperty(ScanHBase.TIME_RANGE_MIN, "0");
+        runner.setProperty(ScanHBase.TIME_RANGE_MAX, "1111111110");
+        runner.setProperty(ScanHBase.LIMIT_ROWS, "10");
+        runner.setProperty(ScanHBase.REVERSED_SCAN, "false");
+        runner.setProperty(ScanHBase.BULK_SIZE, "10");
+
+        runner.enqueue("trigger flow file");
+        runner.run();
+
+        runner.assertTransferCount(ScanHBase.REL_FAILURE, 0);
+        runner.assertTransferCount(ScanHBase.REL_SUCCESS, 1);
+        runner.assertTransferCount(ScanHBase.REL_ORIGINAL, 1);
+
+        MockFlowFile flowFile = runner.getFlowFilesForRelationship(ScanHBase.REL_SUCCESS).get(0);
+        flowFile.assertContentEquals("{\"row\":\"row1\", \"cells\": [" +
```

Comment:
You might want to turn this into a test against two `Map` objects using a JSON parser to evaluate that since things like formatting could hypothetically change between HBase versions.

Reply:
The formatting and structure of JSON representation is coming from NIFI implementation, not HBase. It is defined in:
org.apache.nifi.hbase.io.JsonFullRowSerializer;
org.apache.nifi.hbase.io.JsonQualifierAndValueRowSerializer;

Also, I tried to keep it consistent with TestFetchHBaseRow.